### DISCUSSION TITLE
extract inbuilt plugin only works with encrypted 7z archives

### DISCUSSION THREAD

**Discussion by eyck-dev** [sxyazi/yazi#1462] [1. Q&A]
Is this intentional or a bug on my side? I tried it on encrypted zip and rar files and it doesnt work. I get this error

`
Micro plugin failed:
runtime error: [string "extract"]:1: 7zip exited when extracting '/home/userone/Downloads/gameys/Aero+GPX.rar', error code 2
stack traceback:
	[C]: in function 'error'
	[string "extract"]:1: in upvalue 'fail'
	[string "extract"]:66: in method 'try_with'
	[string "extract"]:21: in function <[string "extract"]:14>
`

**Comment by sxyazi** [comment]
Just tested with `7zz x` and seems like 7zip doesn't support encrypted rar and zip - the `extract` plugin is a wrapper of `7zip`

**Comment by Sohail-XD** [comment]
I tried 7z x and it works. What's the difference between 7z and 7zz?

**Reply by sxyazi** [reply]
See https://askubuntu.com/questions/1465853/difference-between-several-command-line-tools-provided-for-7-zip-compression-li

**Reply by sxyazi** [reply]
I tried using `7z x` on my Linux VM, and it does indeed support encrypted zip files, this should be a bug in Yazi, would you like to file an issue for it?

**Comment by Sohail-XD** [comment]
Why not use p7zip when it has more features.   This is my last question, I won't annoy you any longer 😶

**Reply by sxyazi** [reply]
IIRC p7zip is no longer actively maintained so I have 7zz installed

**Comment by sxyazi** [comment]
Fixed in https://github.com/sxyazi/yazi/issues/1465
